src/Platform/Microsoft.Testing.Platform/TestHostOrcherstrator/TestHostOrchestratorManager.cs

```diff
@@ -31,9 +31,9 @@ public async Task<TestHostOrchestratorConfiguration> BuildAsync(ServiceProvider
             ITestHostOrchestrator orchestrator = factory(serviceProvider);
 
             // Check if we have already extensions of the same type with same id registered
-            if (orchestrators.Any(x => x.Uid == orchestrator.Uid))
+            ITestHostOrchestrator? currentRegisteredExtension = orchestrators.Where(x => x.Uid == orchestrator.Uid).FirstOrDefault();
```

Comment:
We also want to throw a useful localized message if there are more than one results.